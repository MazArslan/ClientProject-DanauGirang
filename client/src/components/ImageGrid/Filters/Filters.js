import React from "react";
import { Row, Col, Input, Button } from "reactstrap";
import propTypes from "prop-types";

/**
 * Filter component
 * Form elements for the filter functionality in the ImageGrid component
 * @param { filterChoice, clearFilters, dateChange, statusChange, applyFilters } props
 */
const filters = props => {
  const {
    filterChoice,
    clearFilters,
    dateChange,
    statusChange,
    applyFilters
  } = props;

  return (
    <React.Fragment>
      <Row>
        <Col xs="3">
          <Row>
            <Col>
              <p>Capture date</p>
            </Col>
          </Row>
          <Row>
            <Col>
              <Input
                type="date"
                name="date"
                placeholder="date placeholder"
                onChange={dateChange}
              />
            </Col>
          </Row>
        </Col>
        <Col xs="3">
          <Row>
            <Col>
              <p>Completion status</p>
            </Col>
          </Row>
          <Row>
            <Col>
              <Input
                type="select"
                name="select"
                value={filterChoice.completionStatus}
                onChange={statusChange}
              >
                <option value="empty">All cards</option>
                <option value="complete">Completed cards</option>
                <option value="incomplete">Incomplete cards</option>
              </Input>
            </Col>
          </Row>
        </Col>
        <Col xs="2">
          {/* Used to add an empty row in order to maintain button positioning */}
          <Row>
            <Col>
              <p>
                <br />
              </p>
            </Col>
          </Row>
          <Row>
            <Col>
              <Button outline color="primary" onClick={applyFilters}>
                Apply filters
              </Button>
            </Col>
          </Row>
        </Col>
        <Col xs="4">
          {/* Used to add an empty row in order to maintain button positioning */}
          <Row>
            <Col>
              <p>
                <br />
              </p>
            </Col>
          </Row>
          <Row>
            <Col>
              <Button outline color="danger" onClick={clearFilters}>
                Clear filters
              </Button>
            </Col>
          </Row>
        </Col>
      </Row>
    </React.Fragment>
  );
};

filters.propTypes = {
  filterChoice: propTypes.object.isRequired,
  clearFilters: propTypes.func.isRequired,
  applyFilters: propTypes.func.isRequired,
  dateChange: propTypes.func.isRequired,
  statusChange: propTypes.func.isRequired
};

export default filters;
