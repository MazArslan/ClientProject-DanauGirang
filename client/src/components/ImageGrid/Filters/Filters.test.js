import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";
import Filters from "./Filters";

Enzyme.configure({ adapter: new Adapter() });

describe("<Filters/>", () => {
  const onActionMock = jest.fn();
  const props = {
    filterChoice: {
      createdOnDate: "2020-03-29",
      completionStatus: "complete"
    },
    clearFilters: onActionMock,
    applyFilters: onActionMock,
    dateChange: onActionMock,
    statusChange: onActionMock
  };
  const container = shallow(<Filters {...props} />);

  describe("component rendering", () => {
    it("should render correctly", () => {
      expect(container.exists()).toBe(true);
      expect(toJson(container)).toMatchSnapshot();
    });

    it("should render date selector", () => {
      expect(
        container
          .find("Input")
          .at(0)
          .props().type
      ).toEqual("date");
    });

    it("should render dropdown selector", () => {
      expect(
        container
          .find("Input")
          .at(1)
          .props().type
      ).toEqual("select");
      expect(
        container
          .find("Input")
          .at(1)
          .props().value
      ).toEqual("complete");
    });

    it("should render options", () => {
      expect(
        container
          .find("option")
          .at(0)
          .props().value
      ).toEqual("empty");
      expect(
        container
          .find("option")
          .at(1)
          .props().value
      ).toEqual("complete");
      expect(
        container
          .find("option")
          .at(2)
          .props().value
      ).toEqual("incomplete");
    });
  });
});
