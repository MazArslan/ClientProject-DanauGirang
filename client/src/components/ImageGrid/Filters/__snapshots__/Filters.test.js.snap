// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filters/> component rendering should render correctly 1`] = `
<Fragment>
  <Row
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      xs="3"
    >
      <Row
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <p>
            Capture date
          </p>
        </Col>
      </Row>
      <Row
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Input
            name="date"
            onChange={[MockFunction]}
            placeholder="date placeholder"
            type="date"
          />
        </Col>
      </Row>
    </Col>
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      xs="3"
    >
      <Row
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <p>
            Completion status
          </p>
        </Col>
      </Row>
      <Row
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Input
            name="select"
            onChange={[MockFunction]}
            type="select"
            value="complete"
          >
            <option
              value="empty"
            >
              All cards
            </option>
            <option
              value="complete"
            >
              Completed cards
            </option>
            <option
              value="incomplete"
            >
              Incomplete cards
            </option>
          </Input>
        </Col>
      </Row>
    </Col>
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      xs="2"
    >
      <Row
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <p>
            <br />
          </p>
        </Col>
      </Row>
      <Row
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Button
            color="primary"
            onClick={[MockFunction]}
            outline={true}
            tag="button"
          >
            Apply filters
          </Button>
        </Col>
      </Row>
    </Col>
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      xs="4"
    >
      <Row
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <p>
            <br />
          </p>
        </Col>
      </Row>
      <Row
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Button
            color="danger"
            onClick={[MockFunction]}
            outline={true}
            tag="button"
          >
            Clear filters
          </Button>
        </Col>
      </Row>
    </Col>
  </Row>
</Fragment>
`;
